src/main/java/net/dv8tion/jda/api/interactions/components/Button.java

```diff
@@ -568,4 +568,40 @@ static Button of(@Nonnull ButtonStyle style, @Nonnull String idOrUrl, @Nonnull E
         Checks.notLonger(idOrUrl, 100, "Id");
         return new ButtonImpl(idOrUrl, "", style, false, emoji);
     }
+
+    /**
+     * Create an enabled button with the provided {@link ButtonStyle style}, URL or ID, label and emoji.
+     
+     * You can use {@link #asDisabled()} to disable it.
+     *
+     * <p>See {@link #link(String, String)} or {@link #primary(String, String)} for more details.
+     *
+     * @param  style
+     *         The button style
+     * @param  idOrUrl
+     *         Either the ID or URL for this button
+     * @param  label
+     *         The text to display on the button
+     * @param  emoji
+     *         The emoji to use as the button label
+     *
+     * @throws IllegalArgumentException
+     *         If any argument is empty or null, the label is longer than 80 characters, the id is longer than 100 characters, or the url is longer than 512 characters
```

Comment:
This should be a list and updated to allow label or emoji to be null, but not both.